### PR TITLE
Better handling of alerts

### PR DESCRIPTION
**Pull request by SparkiDev**
Better checking of encrypted alerts - check unencrypted data size
correctly before using data.
Send an alert if the alert isn't the right size.
Send an alert if the maximum alert count has been reached - but don't
send it if seeing close_notify alert.

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -12788,14 +12789,16 @@ static int DoAlert(WOLFSSL* ssl, byte* input, word32* inOutIdx, int* type,
...
 
-    if (++ssl->options.alertCount >= WOLFSSL_ALERT_COUNT_MAX) {
-        WOLFSSL_MSG("Alert count exceeded");
-        return ALERT_COUNT_E;
-    }
+    if (IsEncryptionOn(ssl, 0))
+        dataSz -= ssl->keys.padSz;
 
     /* make sure can read the message */
-    if (*inOutIdx + ALERT_SIZE > totalSz)
```
Comment (toddouska) [inline src/internal.c]: This seems to be here for a reason.  Why was it removed without explanation?

File: src/internal.c [hunk 2]
```diff
@@ -12806,25 +12809,31 @@ static int DoAlert(WOLFSSL* ssl, byte* input, word32* inOutIdx, int* type,
...
+        WOLFSSL_MSG("\tdecode error");
     }
     if (*type == illegal_parameter) {
-        WOLFSSL_MSG("    illegal parameter");
+        WOLFSSL_MSG("\tillegal parameter");
     }
 #endif
     WOLFSSL_ERROR(*type);
-    if (IsEncryptionOn(ssl, 0)) {
-        if (*inOutIdx + ssl->keys.padSz > totalSz)
```
Comment (toddouska) [inline src/internal.c]: Why was this size check removed?

File: src/internal.c (hunk 1 above)
Comment (SparkiDev) [reply src/internal.c]: Same check just more strict - data size must be ALERT_SIZE.

File: src/internal.c (hunk 2 above)
Comment (SparkiDev) [reply src/internal.c]: Check moved up to start of function.
dataSz = totalSz - *inOutIdx - padSz
dataSz checked to be ALERT_SIZE.

**Pull request merged**